### PR TITLE
chore: Migrating from long-lived credentials to OIDC

### PR DESCRIPTION
**Pull request by alcidesqueiroz**
Closes https://github.com/nearform/fastify-secrets-aws/issues/306

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,11 +20,15 @@ jobs:
...
-        run: npm run test:ci
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ vars.AWS_ROLE_TO_ASSUME }}
+          aws-region: ${{ vars.AWS_REGION }}
+      - run: npm run test:ci
+    permissions:
+      id-token: write
+      contents: read
```
Comment (simoneb) [inline .github/workflows/ci.yml:31]: as a convention, move the permissions at the beginning of the step please

Comment (alcidesqueiroz) [reply .github/workflows/ci.yml]: Just did it. 👍🏼

**Review by simoneb — APPROVED** [review]
LGTM. Please remove the old secrets from the repo settings

File: .github/workflows/ci.yml (hunk 1 above)
Comment (alcidesqueiroz) [reply .github/workflows/ci.yml]: Moved it to the same location used in the `automerge` job

**Pull request merged**